tutorials/tutorial-01.qmd

```diff
@@ -6,33 +6,1403 @@ time: "~45 minutes"
+
+Since we have three layers to inspect, we use `map()` to apply the function to each layer name in `mpa_layers`:
+
+```{r}
+#| label: inspect-mpa-attributes
+#| cache: true
+mpa_descriptions <- map(
+  set_names(mpa_layers),
+  ~ layer_attribute_descriptions(wfs = wfs, layer = .x)
+)
+
+bind_rows(mpa_descriptions, .id = "layer")
+```
+
+#### Identifying Geometry Columns
+
+The `geometry` column in the output flags which attribute holds the spatial data. We extract these for use when downloading:
+
+```{r}
+#| label: get-geom-cols
+get_geom_cols <- function(descriptions) {
+  map_chr(descriptions, ~ .x$name[.x$geometry])
+}
+
+mpa_geom_cols <- get_geom_cols(mpa_descriptions)
+mpa_geom_cols
+```
+
+#### Identifying Filter Candidates
+
+Looking at these descriptions, we can identify attributes useful for filtering:
+
+**Categorical attributes** (character type): useful for selecting specific categories
+
+- **`cddaareas`**: `majorecosy` (ecosystem type) and `designated` (designation status) look promising
+- **`marineprotectedareas`**: `status` likely indicates whether sites are designated or proposed
+
+**Numeric attributes**: useful for threshold-based filtering
+
+- **`natura2000areas`**: `mar_perc` (marine percentage) could help us select predominantly marine sites
+
+#### Exploring Categorical Values
+
+For categorical attributes, we can use `layer_attribute_inspect()` to fetch unique values. This is much cheaper than downloading the full dataset:
+
+```{r}
+#| label: inspect-categorical-filters
+#| cache: true
+# Check ecosystem types in CDDA - we'll want to exclude terrestrial
+layer_attribute_inspect(
+  wfs = wfs,
+  layer = "cddaareas",
+  attribute = "majorecosy"
+)
+
+# Check designation status in CDDA
+layer_attribute_inspect(
+  wfs = wfs,
+  layer = "cddaareas",
+  attribute = "designated"
+)
+
+# Check status values in marine protected areas
+layer_attribute_inspect(
+  wfs = wfs,
+  layer = "marineprotectedareas",
+  attribute = "status"
+)
+```
+
+This tells us:
+
+- **`cddaareas`**: exclude `majorecosy = 'Terrestrial'` and keep only `designated = 'Designated Site'`
+- **`marineprotectedareas`**: keep `status IN ('Designated', 'Designated and managed')`
+
+#### Exploring Numeric Values
+
+For numeric attributes like `mar_perc`, we need to understand the distribution of values to choose an appropriate threshold. The same `layer_attribute_inspect()` function works here too, but returns summary statistics instead of unique values:
+
+```{r}
+#| label: inspect-numeric-filters
+#| cache: true
+layer_attribute_inspect(
+  wfs = wfs,
+  layer = "natura2000areas",
+  attribute = "mar_perc"
+)
+```
+
+The `mar_perc` attribute represents the percentage of each Natura 2000 site that is marine (0-100). The summary shows sites span the full range from fully terrestrial (0%) to fully marine (100%), with a median of ~15% and mean of ~39%.
```

Comment:
do we expect these numbers to change? if so should the text use inline R code to create the numbers?